# Dampier Lab Snakemake Wrappers

A collection of snakemake wrappers that are not suitable for the wider snakemake-wrappers config.


## Usage

The wrappers are designed to be used in the snakemake workflow.
They can be used from the Github repo directly.

```python
rule dorado_duplex:
    input:
        "data/reads.fastq.gz"
    output:
        "data/reads.duplex.fastq.gz"
    wrapper:
        "https://github.com/damlab/damlab-wrappers/blob/main/dorado/duplex"
```

For certain tools, you may need to install the tool environment separately.

```bash
git clone https://github.com/damlab/damlab-wrappers.git
cd damlab-wrappers
make install
```

```python
rule strainline_haplotypes:
    input:
        "data/reads.fastq.gz"
    output:
        "data/haplotypes.fasta"
    params:
        prefix = f"{wrappers_path}/strainline/venv"
    wrapper:
        f"file://{wrappers_path}/strainline/strainline"
```


## Dorado basecalling

This package contains wrappers for the Nanopore dorado tool.

 - [`dorado/duplex`](dorado/duplex/README.md) : Nanopore dorado duplex basecalling tool. Supports GPU acceleration and optional reference-based alignment.
 - [`dorado/simplex`](dorado/simplex/README.md) : Nanopore dorado simplex basecalling tool. Supports GPU acceleration and optional reference-based alignment.
 - [`dorado/demux`](dorado/demux/README.md) : Nanopore dorado demultiplexing tool. Supports various barcoding kits and custom barcode arrangements.
 - more to come as needed...

## POD5

This package contains wrappers for the POD5 file format.

 - [`pod5/convert_fast5`](pod5/convert_fast5/README.md) : Convert Oxford Nanopore FAST5 files to the newer POD5 format.
 - [`pod5/split_by_channel`](pod5/split_by_channel/README.md) : Split POD5 files by channel.


## Multiple sequence alignment

This package contains wrappers for the multiple sequence alignment tools.

 - [`MSA/muscle`](MSA/muscle/README.md) : MUSCLE multiple sequence alignment tool.

## Strainline

This package contains wrappers for the Strainline reference free haplotype assembly tool.
This package requires the `strainline` tool to be installed via the provided makefile.

 - [`strainline/strainline`](strainline/strainline/README.md) : Strainline haplotype assembly tool.
 - [`strainline/clipqs`](strainline/clipqs/README.md) : ClipQS tool to orient and clip sequences generated by the Strainline tool.

## Cigarmath

This package contains wrappers for the Cigarmath library.

 - [`cigarmath/deletion_frequency`](cigarmath/deletion_frequency/README.md) : Calculate the deletion frequency of a given region.

## Phylogenetic tree construction

This package contains wrappers for the Phylogenetic tree construction tools.

 - [`phylo/FastTree`](phylo/FastTree/README.md) : Fasttree phylogenetic tree construction tool.
 - [`phylo/phytreeviz`](phylo/phytreeviz/README.md) : Phytreeviz tool for visualizing phylogenetic trees.
 - [`phylo/reroot`](phylo/reroot/README.md) : Rerooting tool to reroot a tree using dendropy.
 
## Visualization

A collection of snakemake wrappers of Python APIs for visualization.

- [`visualization/jaspar2logo`](visualization/jaspar2logo/README.md) : Create sequence logos from frequency counts.

## Miscellaneous

- [`picard/addorreplacereadgroups`](picard/addorreplacereadgroups/README.md) : Picard tool to add or replace read groups in a BAM file. Slightly modified to have an easier way to set parameters.

## Testing

Use `make test` to check all tests.

