from os.path import join

wd = workflow.basedir
TEST_DATA_DIR = join(wd, '../../tests')

rule all:
    input:
        "test_bam/sample3.bam",
        "test_bam/sample5.bam",
        "test_bam/missing.bam",
        "test_fastq/sample3.fastq",
        "test_fastq/sample5.fastq",
        "test_fastq/missing.fastq"
    conda: 'env.yaml'
    shell: "pytest tests.py"


rule test_demux_bam:
    input:
        reads = 'basecalled.bam'
    output:
        directory("test_bam")
    params:
        kit_name = "SQK-NBD114-24",
        barcode_to_output = {
            "barcode03": "sample3.bam",  # maps to sample3.bam
            "barcode05": "sample5.bam",  # maps to sample5.bam
            "barcode99": "missing.bam"   # should create empty file
        }
    threads: 2
    wrapper:
        "file:../" 

rule test_demux_fastq:
    input:
        reads = 'basecalled.bam'
    output:
        directory("test_fastq")
    params:
        kit_name = "SQK-NBD114-24",
        emit_fastq = True,
        barcode_to_output = {
            "barcode03": "sample3.fastq",  # maps to sample3.fastq
            "barcode05": "sample5.fastq",  # maps to sample5.fastq
            "barcode99": "missing.fastq"   # should create empty file
        }
    threads: 2
    wrapper:
        "file:../" 